test/uri/uri_user_info_test.cc

```diff
@@ -0,0 +1,22 @@
+
+#include <gtest/gtest.h>
+#include <sourcemeta/jsontoolkit/uri.h>
+
+TEST(URI_user_info, working_example) {
+  sourcemeta::jsontoolkit::URI uri{"http://user:@host:80/"};
+  EXPECT_TRUE(uri.user_info().has_value());
+  EXPECT_EQ(uri.user_info().value(), "user:");
+}
+
+// Note: Using the format "user:password" is depreacted.
+// But we would like to support it for users that would still use it.
+TEST(URI_user_info, depreacted_example) {
+  sourcemeta::jsontoolkit::URI uri{"http://user:password@host:80/"};
+  EXPECT_TRUE(uri.user_info().has_value());
+  EXPECT_EQ(uri.user_info().value(), "user:password");
+}
```

Comment:
I wonder if we should not be strict on this? or at least print a warning.

Reply:
I think it's fine. Even if it's deprecated, is part of the RFC. The warning you put in the docs should be enough!